### PR TITLE
Custom logo support

### PR DESCRIPTION
**Pull request by dmitrijs-voronovs**
- Added FE support for custom logos
Default size is `150x40`

Ticket #SWPWA-153

### CODE REVIEW

File: src/app/component/Header/Header.component.js [hunk 1]
```diff
@@ -308,15 +327,20 @@ export default class Header extends PureComponent {
...
-                    </g>
-                </svg>
+                { header_logo_src
+                    ? <img src={ `/media/logo/${header_logo_src}` } style={ logoSize } alt={ logo_alt } />
+                    : (
```
Comment (alfredsgenkins) [inline src/app/component/Header/Header.component.js]: Please do not render our SVG icon. Render blank space.

File: src/app/component/Header/Header.component.js [hunk 2]
```diff
@@ -308,15 +327,20 @@ export default class Header extends PureComponent {
...
-                    </g>
-                </svg>
+                { header_logo_src
+                    ? <img src={ `/media/logo/${header_logo_src}` } style={ logoSize } alt={ logo_alt } />
```
Comment (alfredsgenkins) [inline src/app/component/Header/Header.component.js]: Please do not use conditional rendering, create a function for that.

File: src/app/store/Config/Config.action.js [hunk 3]
```diff
@@ -9,8 +9,14 @@
  * @link https://github.com/scandipwa/base-theme
  */
 export const UPDATE_CONFIG = 'UPDATE_CONFIG';
+export const SET_LOADING_STATUS = 'SET_LOADING_STATUS';
```
Comment (alfredsgenkins) [inline src/app/store/Config/Config.action.js]: How is that needed? Why? Just either show blank or logo if it is loaded.

File: src/app/store/Config/Config.dispatcher.js [hunk 4]
```diff
@@ -25,6 +25,7 @@ export class ConfigDispatcher extends QueryDispatcher {
         if (data) {
             BrowserDatabase.setItem(data, 'config', ONE_MONTH_IN_SECONDS);
             dispatch(updateConfig(data));
+            dispatch(setLoadingStatus(false));
```
Comment (alfredsgenkins) [inline src/app/store/Config/Config.dispatcher.js]: Even knowing that this is redundant (isLoading part), why to create a separate action for this? Just set the loading to false when updaing the config.

File: src/app/store/Config/Config.action.js (hunk 3 above)
Comment (dmitrijs-voronovs) [reply src/app/store/Config/Config.action.js]: According to the ticket, 
> Default logo - ScandiPWA. If there is option to delete logo, then our logo will show as a placeholder.
Should I make it blank anyway?

File: src/app/store/Config/Config.reducer.js [hunk 5]
```diff
@@ -12,8 +12,18 @@
...
 
+export const ALWAYS_UPDATE_IN_CONFIG = ['header_logo_src', 'logo_height', 'logo_width'];
+
 export const filterStoreConfig = config => Object.entries(config).reduce(
-    (acc, [key, value]) => (value ? { ...acc, [key]: value } : acc),
+    (acc, [key, value]) => {
+        if (ALWAYS_UPDATE_IN_CONFIG.includes(key)) {
+            return value
+                ? { ...acc, [key]: value }
+                : { ...acc, [key]: undefined };
```
Comment (dmitrijs-voronovs) [inline src/app/store/Config/Config.reducer.js]: Explicitly set to undefined as empty width / height arrives as null

Comment (dmitrijs-voronovs) [inline src/app/store/Config/Config.reducer.js]: This way we make sure, that logo changes immediately on FE after it was changed on BE.
Explanation:
- Before
In browser database: { logo_height : 144, header_logo_src: '124124' }
In action for reducer: { logo_width : 411 }
Reducer returns merged objects: { logo_height : 144, header_logo_src: '124124', logo_width : 411 } , which is incorrect, as we are handling no-url case, no-width case and no-height case.
- Now:
After browser gets updated, problem solves, as BrowserDatabase already got updated with { logo_width : 411 } values (BrowserDatabase is not getting merged, just gets overwritten). Now reducer returns { logo_width : 411 } + { logo_width : 411 } = { logo_width : 411 } with no additional incorrect values.

**Comment by alfredsgenkins** [comment]
Merged as a part of #354

**Pull request closed without merging**